### PR TITLE
[RISCV] Add Ssqosid support to -march.

### PR DESCRIPTION
**Pull request by topperc**

### CODE REVIEW

File: clang/test/Preprocessor/riscv-target-features.c [hunk 1]
```diff
@@ -1612,6 +1613,14 @@
 // RUN:   -o - | FileCheck --check-prefix=CHECK-SUPM-EXT %s
 // CHECK-SUPM-EXT: __riscv_supm 8000{{$}}
 
+// RUN: %clang --target=riscv32 -menable-experimental-extensions \
+// RUN:   -march=rv32i_ssqosid1p0 -E -dM %s \
+// RUN:   -o - | FileCheck --check-prefix=CHECK-SSQOSID-EXT %s
+// RUN: %clang --target=riscv64 \
+// RUN:   -march=rv64i_ssqosid1p0 -E -dM %s -menable-experimental-extensions \
```
Comment (wangpc-pp) [inline clang/test/Preprocessor/riscv-target-features.c:1620]: Why is `-menable-experimental-extensions` here?

Comment (topperc) [reply clang/test/Preprocessor/riscv-target-features.c:1620]: It's not ratified yet right?

Comment (dtcxzyw) [reply clang/test/Preprocessor/riscv-target-features.c:1620]: It is required for clang when using experimental RISC-V extensions.
> def menable_experimental_extensions : Flag<["-"], "menable-experimental-extensions">, Group<m_Group>,
  HelpText<"Enable use of experimental RISC-V extensions.">;

Comment (wangpc-pp) [reply clang/test/Preprocessor/riscv-target-features.c:1620]: Oh sorry, I meant why `-menable-experimental-extensions` is in line 1620, not line 1619.

Comment (topperc) [reply clang/test/Preprocessor/riscv-target-features.c:1620]: Because I copied it from supm which has the same issue.

**Review by asb — APPROVED** [review]
LGTM.

**Pull request merged**